mlir/lib/Target/SPIRV/Serialization/SerializeOps.cpp

```diff
@@ -311,6 +311,15 @@ LogicalResult Serializer::processFuncOp(spirv::FuncOp op) {
     op.addEntryBlock();
     if (failed(processFuncParameter(op)))
       return failure();
+
+    // Erasing the body of the function destroys arguments, so we need to remove
+    // them from the map to avoid problems when processing invalid values used
+    // as keys. We have already serialized function arguments so we probably can
+    // remove them from the map as external function will not have any uses.
+    for (Value arg : op.getArguments())
+      if (valueIDMap.count(arg))
```

Comment:
IIRC this is a DenseMap
```suggestion
      if (valueIDMap.contains(arg))
```